src/editor/mod.rs

```diff
@@ -558,3 +577,20 @@ pub fn start_editor(event_loop_proxy: EventLoopProxy<UserEvent>) -> NeovimHandle
     });
     handler
 }
+
+/// Get proper dark(white text)/light(black text) theme of window for background color.
+/// Based on formula in https://graphicdesign.stackexchange.com/questions/62368/automatically-select-a-foreground-color-based-on-a-background-color
+#[cfg(target_os = "macos")]
+fn window_theme_for_background(background_color: Option<Color4f>) -> Option<Theme> {
+    if let Some(background_color) = background_color {
+        if 0.2126 * background_color.r + 0.7152 * background_color.g + 0.0722 * background_color.b
+            > 0.5
+        {
+            Some(Theme::Light)
+        } else {
+            Some(Theme::Dark)
+        }
+    } else {
+        None
+    }
+}
```

Comment:
instead of nested if/else, this would be more legible. At least creating a function that self describes what it is doing by their annotation would be great. for example:

```rust
/// Based on formula in https://graphicdesign.stackexchange.com/questions/62368/automatically-select-a-foreground-color-based-on-a-background-color
/// Checks if the color is light.
#[cfg(target_os = "macos")]
fn is_light_theme(color: &Color4f) -> bool {
    0.2126 * color.r + 0.7152 * color.g + 0.0722 * color.b > 0.5
}

/// Get proper dark(white text)/light(black text) theme of window for background color.
#[cfg(target_os = "macos")]
fn window_theme_for_background(background_color: Option<Color4f>) -> Option<Theme> {
    background_color?;

    match background_color.unwrap() {
        color if is_light_theme(&color) => Some(Theme::Light),
        _ => Some(Theme::Dark),
    }
}
```